### PR TITLE
Fast define

### PR DESCRIPTION
**Pull request by minggo**
Refactor current implementation of decorator codes with fast define.

### CODE REVIEW

**Comment by minggo** [comment]
@cocos-robot run test cases

**Comment by jareguo** [comment]
目前能做到比较理想的估计也就这条路了。如果可以的话，原生运行时完全剔除默认值信息也行，这样更彻底一些。不过 3.7 编辑器也原生化之后，感觉 U_PROPERTY 之类的东西还是跑不掉。

**Comment by minggo** [comment]
@cocos-robot run test cases

File: cocos/core/data/decorators/property.ts [hunk 1]
```diff
@@ -248,6 +248,12 @@ function setDefaultValue<T> (
     propertyKey: PropertyKey,
     descriptor: BabelPropertyDecoratorDescriptor | undefined,
 ) {
+    // Default values are needed by editor, and now editor run with web version, so don't
```
Comment (shrinktofit) [inline cocos/core/data/decorators/property.ts:251]: A more suitable way to do this is:

```diff
// cocos/core/data/decorators/editable.ts

+++ const EDITOR_DECORATOR_ENABLED = !DEV && !NATIVE;

--- export const menu: (path: string) => ClassDecorator =    DEV ? makeEditorClassDecoratorFn('menu') : emptyDecoratorFn;
+++ export const menu: (path: string) => ClassDecorator =    EDITOR_DECORATOR_ENABLED ? makeEditorClassDecoratorFn('menu') : emptyDecoratorFn;
```

**Review by shrinktofit — APPROVED** [review]

**Comment by suqionglian** [comment]
@cocos-robot run test cases

**Comment by minggo** [comment]
As this PR will bring some issues, should not merge it now.

**Comment by minggo** [comment]
I tested it, it worked fine.

File: cocos/core/assets/texture-cube.jsb.ts [hunk 2]
```diff
@@ -292,6 +252,10 @@ textureCubeProto._deserialize = function (serializedData: ITextureCubeSerializeD
...
 }
 
-clsDecorator(TextureCube);
-
 legacyCC.TextureCube = jsb.TextureCube;
+
+// handle meta data, it is generated automatically
+const TextureCubeProto = TextureCube.prototype;
+serializable(TextureCubeProto, 'isRGBE');
+serializable(TextureCubeProto, '_mipmaps');
```
Comment (dumganhar) [inline cocos/core/assets/texture-cube.jsb.ts:260]: Miss serializable `_mipmapAtlas` & `_mipmapMode`

File: cocos/core/root.ts [hunk 3]
```diff
@@ -490,9 +490,7 @@ export class Root {
 
             legacyCC.director.emit(legacyCC.Director.EVENT_BEFORE_COMMIT);
             cameraList.sort((a: Camera, b: Camera) => a.priority - b.priority);
-            if (!SERVER_MODE) {
```
Comment (dumganhar) [inline cocos/core/root.ts]: Why delete this check? No need anymore?

Comment (minggo) [reply cocos/core/root.ts]: The upstream codes already modified it. I don't know why it also has this modification here. I don't change it.

**Review by SantyWang — APPROVED** [review]

File: cocos/core/assets/texture-cube.jsb.ts (hunk 2 above)
Comment (minggo) [reply cocos/core/assets/texture-cube.jsb.ts:260]: Good catch. I added them.

**Pull request merged**